envi/__init__.py

```diff
@@ -156,6 +156,26 @@ def archGetRegisterGroups(self):
         allr = [rname for rname in regctx.getRegisterNames()]
         return [ ('all', allr), ]
 
+    def archModifyFuncAddr(self, va, arch):
```

Comment:
It seems a bit odd that we are passing arch in as an argument to the architecture module itself.  If we wanted to swtch archs during code flow ( arm stylez ) it's our current arch ( this module ) that will be deciding it.  With my comments below (about info dict) any arch would be free to include 'arch':<newarch> in the info dict ( or do nothing to assume it is unchanged )